### PR TITLE
[Runtime] Enable the -Wsign-conversion on monotouch-debug files.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Enable the flag that was disabled via pragmas and fix warnings.

Continuation of PR: https://github.com/xamarin/xamarin-macios/pull/7405

### CODE REVIEW

**Comment by mandel-macaque** [comment]
Known issue: https://github.com/xamarin/maccore/issues/581

File: runtime/monotouch-debug.m [hunk 1]
```diff
@@ -837,7 +833,8 @@ static int sdb_recv (void *buf, int len)
 			connection.ip = [ips objectAtIndex: i];
 			connection.uniqueRequest = unique_request;
 			[connections addObject: connection];
-			[connection connect: [ips objectAtIndex: i] port: monodevelop_port completionHandler: ^void (bool success)
+			assert (monodevelop_port > 0); // Make sure we do have a valid port
```
Comment (rolfbjarne) [inline runtime/monotouch-debug.m]: Don't assert, instead print something (NSLog) and then just not try to connect.

File: runtime/monotouch-debug.m [hunk 2]
```diff
@@ -1518,7 +1515,7 @@ int monotouch_debug_connect (NSMutableArray *ips, int debug_port, int output_por
 		ip = [[ips objectAtIndex:i] UTF8String];
 		
 		memset (sockaddr, 0, sizeof (sockaddr));
-		
+
```
Comment (rolfbjarne) [inline runtime/monotouch-debug.m]: whitespace change.

File: runtime/monotouch-debug.m [hunk 3]
```diff
@@ -1725,7 +1722,4 @@ int monotouch_debug_connect (NSMutableArray *ips, int debug_port, int output_por
 
 #else
 int xamarin_fix_ranlib_warning_about_no_symbols_v2;
-#endif /* DEBUG */
-
-
-#pragma clang diagnostic pop
+#endif /* DEBUG */
```
Comment (rolfbjarne) [inline runtime/monotouch-debug.m]: Missing newline at end of file.

File: runtime/monotouch-debug.m [hunk 4]
```diff
@@ -837,7 +833,12 @@ static int sdb_recv (void *buf, int len)
 			connection.ip = [ips objectAtIndex: i];
 			connection.uniqueRequest = unique_request;
 			[connections addObject: connection];
-			[connection connect: [ips objectAtIndex: i] port: monodevelop_port completionHandler: ^void (bool success)
+			if (monodevelop_port > 0) {
```
Comment (rolfbjarne) [inline runtime/monotouch-debug.m]:
```suggestion
			if (monodevelop_port <= 0) {
```

**Comment by mandel-macaque** [comment]
Known issues: https://github.com/xamarin/maccore/issues/581

**Pull request merged**